frontend/src/components/Options.vue

```diff
@@ -1,38 +1,107 @@
 
-      <b-dropdown-item @click="toggleRewards()">Reward Bar: {{ hideRewards ? 'Off' : 'On' }}</b-dropdown-item>
+        <b-dropdown-item @click="toggleRewards()">Reward Bar: {{ hideRewards ? 'Off' : 'On' }}</b-dropdown-item>
 
-      <b-dropdown-item @click="toggleAdvanced()">Advanced UI: {{ hideAdvanced ? 'Off' : 'On' }}</b-dropdown-item>
-    </b-nav-item-dropdown>
-  </b-navbar-nav>
+        <b-dropdown-item @click="toggleAdvanced()">Advanced UI: {{ hideAdvanced ? 'Off' : 'On' }}</b-dropdown-item>
+
+        <b-dropdown-item @click="claimSkill(0)">Claim Skill </b-dropdown-item>
+
+      </b-nav-item-dropdown>
+    </b-navbar-nav>
+
+    <b-modal class="centered-modal" ref="need-gas-modal" title="Need Gas?"
+      @ok="claimSkill(1)" ok-title="Next" @cancel="$router.push('/portal')" cancel-title="Go to WAX Bridge" >
+        Need Gas? Try our WAX Bridge, which will pay you .5% under market rate to sell your WAX for BNB!
+    </b-modal>
+    <b-modal class="centered-modal" ref="stake-suggestion-modal" title="Stake Skill"
+      @ok="$router.push('/stake')" ok-only ok-title="Go to Stake" >
+        If you stake your SKILL now, we will give you a 10% bonus in SKILL that you can use in-game right away!
+      <a href="#" @click="claimSkill(2)"> <br>No thanks, I'd rather {{ (this.currentWithdrawTax > 0)?"pay " +
+        this.formattedTaxAmount + " in taxes and " : ""  }}forfeit my bonus </a>
+    </b-modal>
+    <b-modal class="centered-modal" ref="claim-confirmation-modal" title="Claim Skill" ok-title="I am sure"
+      @ok="onClaimTokens()"> You are about to {{ (this.currentWithdrawTax > 0)?"pay " + this.currentWithdrawTax +
+      " tax for early withdrawal, costing you " + this.formattedTaxAmount + " SKILL. You will also " : "" }}
+      forfeit all bonus SKILL earnings for 3 days, costing {{formattedBonusLost}} bonus SKILL. Are you sure
+      you wish to continue? <b>This action cannot be undone.</b>
+    </b-modal>
+  </div>
 </template>
 
-<script>
+<script lang="ts">
 import Events from '../events';
+import { mapActions, mapState } from 'vuex';
+import BN from 'bignumber.js';
+import Web3 from 'web3';
+import { Accessors } from 'vue/types/options';
+import { BootstrapVueIcons } from 'bootstrap-vue';
+import Vue from 'vue';
+
+Vue.use(BootstrapVueIcons);
+interface StoreMappedState {
+  skillRewards: string;
+  directStakeBonusPercent: number;
+}
 
-export default {
+interface StoreMappedActions {
+  claimTokenRewards(): Promise<void>;
+}
+interface Data {
+  showGraphics: boolean;
+  hideRewards: boolean;
+  hideAdvanced: boolean;
+  currentWithdrawTax: number;
+  directStakeBonusPercent: number;
+}
+
+export default Vue.extend({
   created() {
     this.showGraphics = localStorage.getItem('useGraphics') === 'true';
     this.hideRewards = localStorage.getItem('hideRewards') === 'true';
     this.hideAdvanced = localStorage.getItem('hideAdvanced') === 'true';
+    this.currentWithdrawTax = 0.14;
```

Comment:
is this able to be gotten dynamically, or is it just for display purposes?

Reply:
Pulling from store now, thanks